solidity/contracts/bridge/Bridge.sol

```diff
@@ -32,89 +36,232 @@ pragma solidity 0.8.4;
 ///         about the sweep increasing appropriate balances in the Bank.
 /// @dev Bridge is an upgradeable component of the Bank.
 contract Bridge {
-    struct DepositInfo {
-        uint64 amount;
+    // TODO: Consider using a custom fork of Summa libs adjusted to Solidity 8.
+    using BTCUtils for bytes;
+    using BytesLib for bytes;
+
+    /// @notice Represents Bitcoin transaction data as described in:
+    ///         https://developer.bitcoin.org/reference/transactions.html#raw-transaction-format
+    struct TxInfo {
+        // Transaction version number (4-byte LE).
+        bytes4 version;
+        // All transaction inputs prepended by the number of inputs encoded
+        // as a VarInt. Single vector item looks as follows:
+        // https://developer.bitcoin.org/reference/transactions.html#txin-a-transaction-input-non-coinbase
+        // though SegWit inputs don't contain the signature script (scriptSig).
+        // All encoded input transaction hashes are little-endian.
+        bytes inputVector;
+        // All transaction outputs prepended by the number of outputs encoded
+        // as a VarInt. Single vector item looks as follows:
+        // https://developer.bitcoin.org/reference/transactions.html#txout-a-transaction-output
+        bytes outputVector;
+        // Transaction locktime (4-byte LE).
+        bytes4 locktime;
+    }
+
+    /// @notice Represents data which must be revealed by the depositor during
+    ///         deposit reveal.
+    struct RevealInfo {
+        // Index of the funding output belonging to the funding transaction.
+        uint8 fundingOutputIndex;
+        // Ethereum depositor address.
+        address depositor;
+        // The blinding factor as 8 bytes. Byte endianness doesn't matter
+        // as this factor is not interpreted as uint.
+        bytes8 blindingFactor;
```

Comment:
Would it be possible to increase it? Not in this PR and we need to do some math first but just ensuring it's easily doable later. It looks like just a change in `expectedScript` computation + unit tests, correct?

Reply:
Do you mean the byte size? If so, yeah we can increase but we need to remember we'll need to do the same on `tbtc-ts` side to stay consistent.

Reply:
Captured in https://github.com/keep-network/tbtc-v2/issues/102.